### PR TITLE
Week 3: Step 2

### PR DESCRIPTION
**Pull request by cxfloeder**
Used fetch() to get a message from the server and add it to the DOM.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -12,6 +12,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** Uses fetch() to add a message to the DOM */
+ function getMessage() {
+    // fetches from the DataServlet
+    const responsePromise = fetch("/data");
+
+    // passes the response to handleResponse after the request is complete
```
Comment (ianjjohnson) [inline portfolio/src/main/webapp/script.js:20]: Nit: capitalize start of comments (throughout)

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -12,6 +12,39 @@
...
+ */
+ function handleResponse(response) {
+    const textPromise = response.text();
+
+    textPromise.then(addMessageToDom);
+ }
+
+/** Adds the message to the DOM */
+function addMessageToDom(message) {
+
```
Comment (ianjjohnson) [inline portfolio/src/main/webapp/script.js:36]: nit: remove extra blank line

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -12,6 +12,39 @@
...
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+ function handleResponse(response) {
+    const textPromise = response.text();
+
+    textPromise.then(addMessageToDom);
+ }
+
+/** Adds the message to the DOM */
```
Comment (ianjjohnson) [inline portfolio/src/main/webapp/script.js:34]: Nit (throughout): end doc comments with a period

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -12,6 +12,39 @@
...
+function addMessageToDom(message) {
+
+    const messageContainer = document.getElementById('message_container');
+    messageContainer.innerText = message;
+}
+
+/** Practices using arrow functions to shorten the code */
```
Comment (ianjjohnson) [inline portfolio/src/main/webapp/script.js:41]: Lambda functions (like you're using here) are super common in JS and a nice way to condense some simple code.
Nice! 👍

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (cxfloeder) [reply portfolio/src/main/webapp/script.js]: fixed

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -12,6 +12,37 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** Uses fetch() to add a message to the DOM. */
+ function getMessage() {
+    // Fetches from the DataServlet.
+    const responsePromise = fetch("/data");
```
Comment (jakecarlson1) [inline portfolio/src/main/webapp/script.js:18]: You may want to put "/data" in a constant (MESSAGE_API_URL or something like that). This way, if you ever rename this endpoint, you only have to update the path in one location.

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -12,6 +12,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+const MESSAGE_API_URL = "/data";
+
+/** Uses fetch() to add a message to the DOM. */
+ function getMessage() {
+    // Fetches from the DataServlet.
+    const responsePromise = fetch(MESSAGR_API_URL);
```
Comment (jakecarlson1) [inline portfolio/src/main/webapp/script.js:20]: s/MESSAGR/MESSAGE/

**Pull request merged**